src/containers/course-section/resource-item/ResourceItem.tsx

```diff
@@ -1,23 +1,58 @@
-import { FC, ReactElement } from 'react'
+import { FC, ReactElement, useEffect } from 'react'
 import Box from '@mui/material/Box'
 import IconButton from '@mui/material/IconButton'
 import CloseIcon from '@mui/icons-material/Close'
+import { DatePicker } from '@mui/x-date-pickers/DatePicker'
+import TextField from '@mui/material/TextField'
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns'
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
 
+import AppSelect from '~/components/app-select/AppSelect'
 import IconExtensionWithTitle from '~/components/icon-extension-with-title/IconExtensionWithTitle'
+import { useResourceAvailabilityContext } from '~/context/resources-availability-context'
+import openIcon from '~/assets/img/cooperation-details/resource-availability/open-icon.svg'
+import openFrom from '~/assets/img/cooperation-details/resource-availability/open-from.svg'
+import closedIcon from '~/assets/img/cooperation-details/resource-availability/closed-icon.svg'
 
+import {
+  CourseResources,
+  ResourceAvailabilityStatus,
+  ResourcesAvailability,
+  ResourcesTabsEnum as ResourcesTypes
+} from '~/types'
+
+import { selectionFields } from '~/containers/course-section/resource-item/ResourceItem.constants'
 import { resourcesData } from '~/containers/course-section/CourseSectionContainer.constants'
 import { styles } from '~/containers/course-section/resource-item/ResourceItem.styles'
-import { ResourcesTabsEnum as ResourcesTypes, CourseResources } from '~/types'
 
 interface ResourceItemProps {
   resource: CourseResources
   deleteResource: (resource: CourseResources) => void
+  setResourceAvailability: (
+    id: string,
+    availability: ResourceAvailabilityStatus,
+    openFromDate: string | null
+  ) => void
+}
+
+const availabilityIcons = {
```

Comment:
will we use these icons in other files in the future? If yes, makes sense to move them outside and import

Reply:
Not expected